### PR TITLE
Added AppImage generation step. Added / changed some steps to make ev…

### PR DESCRIPTION
**Pull request by XandrosDarkstorm**
## Required sign-off

- [x] I confirm that my PR does not contain any commercial or protected assets and/or source code.
- [x] I agree in advance that my codes will be licensed automatically under the Apache License or similar BSD/MIT-like
      open source licenses in case if OpenNox Project will adopt such a non-GPL license in the future.

### PR explanation

This PR changes workflow file to accomodate automated AppImage assembly for main OpenNox executables. Please accept [this PR](https://github.com/noxworld-dev/build-docker/pull/1), before accepting this one.

### CODE REVIEW

**Comment by Evengard** [comment]
@dennwc I'll take care of that during the weekend

**Comment by dennwc** [comment]
@Evengard any update on it? Should I look into it instead?

**Comment by Evengard** [comment]
The weekend still hasn't ended, I'm still planning to check it out today =)

File: .github/workflows/build-and-release.yml [hunk 1]
```diff
@@ -212,10 +212,35 @@ jobs:
...
+    if: startsWith(github.ref, 'refs/tags/') || (github.ref == 'refs/heads/appimage-workflow-test') || (github.ref == 'refs/heads/ci_tests')
+    needs: [linux_docker]
+    steps:
+      - uses: actions/download-artifact@v3
+        name: Retrieving compiled OpenNox
+        with:
+          name: opennox-linux
+          path: build
+
+      - uses: docker://ghcr.io/xandrosdarkstorm/docker-build:latest-linux-packaging-appimage
```
Comment (Evengard) [inline .github/workflows/build-and-release.yml]: That should be changed to the noxworld-dev environment before committing. Update it when the related PR in docker-build repo will be merged.

**Comment by Evengard** [comment]
Seems to be fine except the path to the actual docker image.

**Comment by XandrosDarkstorm** [comment]
All locations for Docker images are pointing to noxworld-dev. This should resolve the problem.

**Pull request merged**